database-enclave/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -0,0 +1,121 @@
+ * by conclave.
+ */
+@CommandLine.Command(name = "database-client",
+        mixinStandardHelpOptions = true,
+        description = "Simple client that communicates with the DatabaseEnclave using the web host.")
+public class Client implements Callable<Void> {
+
+    private static final Logger logger = LoggerFactory.getLogger(Client.class);
+
+    @CommandLine.Option(names = {"-o", "--command"}
+            , description = "Command Options: ${CREATE/ADD/VERIFY}")
+    private CommandType commandType = null;
+
+    //Use picocli to provide command line parameters
+    @CommandLine.Parameters(description = "username password")
+    private final List<String> userDetails = new ArrayList<>();
+
+    @CommandLine.Option(names = {"-u", "--url"},
+            required = true,
+            description = "URL of the web host running the enclave.")
+    private String url;
+
+    @CommandLine.Option(names = {"-c", "--constraint"},
+            required = true,
+            description = "Enclave constraint which determines the enclave's identity and whether it's acceptable to use.",
+            converter = EnclaveConstraintConverter.class)
+    private EnclaveConstraint constraint;
+
+    @Override
+    public Void call() throws Exception {
+        /*A new private key is generated. Enclave Client is created using this private key and constraint.
+        A corresponding public key will be used by the enclave to encrypt data to be sent to this client*/
+        EnclaveClient enclaveClient = new EnclaveClient(constraint);
+
+        try (WebEnclaveTransport transport = new WebEnclaveTransport(url);
+             EnclaveClient client = enclaveClient) {
+            /*Retrieve the enclaveInstanceInfo object, i.e. the remote attestation object and verify it against the
+            constraint*/
+            client.start(transport);
+
+            //Collect merchants and service providers credit card numbers list
+            InputData inputData = getInputData();
+
+            byte[] requestMailBody = serialize(inputData);
+
+            //Client will send its credit card list to enclave and wait for other client to send their list
+            EnclaveMail responseMail = client.sendMail(requestMailBody);
+
+            //ResponseMail is null till enclave doesn't reply back to the client
+            if (responseMail == null) {
+                do {
+                    Thread.sleep(2000);
+                    //Poll for reply to enclave
+                    responseMail = enclaveClient.pollMail();
+                } while (responseMail == null);
+            }
+            logger.info("Enclave Reply is : " + new String(responseMail.getBodyAsBytes()));
+        }
+        return null;
+    }
+
+    private InputData getInputData() {
+        InputData inputData = new InputData();
+        inputData.setCommandType(commandType);
+        if(!userDetails.isEmpty()) {
+            if(commandType == CommandType.ADD) {
```

Comment:
I notice some formatting issues, we might need a cleanup to the files. For example, `if(` needs a space `if (`

Reply:
done